hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2240,14 +2240,39 @@ public void deleteKey(OmKeyArgs args) throws IOException {
     }
   }
 
-  // TODO: HDDS-2418 - Complete stub below for server logic
   @Override
   public List<RepeatedOmKeyInfo> listTrash(String volumeName,
       String bucketName, String startKeyName, String keyPrefix, int maxKeys)
       throws IOException {
 
-    List<RepeatedOmKeyInfo> deletedKeys = new ArrayList<>();
-    return deletedKeys;
+    if (isAclEnabled) {
+      checkAcls(ResourceType.BUCKET, StoreType.OZONE, ACLType.LIST,
+          volumeName, bucketName, keyPrefix);
+    }
+
+    boolean auditSuccess = true;
+    Map<String, String> auditMap = buildAuditMap(volumeName);
+    auditMap.put(OzoneConsts.BUCKET, bucketName);
+    auditMap.put(OzoneConsts.START_KEY, startKeyName);
+    auditMap.put(OzoneConsts.KEY_PREFIX, keyPrefix);
+    auditMap.put(OzoneConsts.MAX_KEYS, String.valueOf(maxKeys));
+
+    try {
+      metrics.incNumKeyLists();
```

Comment:
Wondering if we should create a new metric `NumTrashKeyLists` & `NumTrashKeyListFails` for sake of granularity.

Reply:
That makes sense to me.  I created HDDS-2484 to follow up with that.